syntax = "proto3";

option java_package = "com.nk.tripleled.proto";

message AllColorConfig
{
    oneof Settings {
        ColorRGB colorRGB = 1;
        ColorHSV colorHSV = 2;
        uint32 brightness = 3;
    };
};

message ColorHSV
{
    uint32 H = 1;
    uint32 S = 2;
    uint32 V = 3;
};

message ColorRGB
{
    uint32 R = 1;
    uint32 G = 2;
    uint32 B = 3;
};